### PR TITLE
impr: update styles in login form components

### PR DESCRIPTION
**Pull request by sidarth-23**
## Summary
- Updated the styles for the main container of the login form to use a different background color, padding, and spacing.
- Updated the styles for the container of the form fields to use a different spacing between the elements.
- Updated the styles for the text inside the form fields to use a different text color.
- Added a new dependency on the "lucide-react" library to be able to use icons in the buttons.
- Updated the styles for the buttons to include a gap between the icon and the text, and added support for a start icon.
- Updated the styles for the labels to include support for different variants and added support for displaying field errors.

### Checklist
- [x] Tested (Must)
- [ ] Test Case added
- [x] Build Successful (Must)
- [x] Sufficient Code comments added (Must)
- [ ] Attached Screenshots / Videos 
- [x] All Relevant Documents added

### CODE REVIEW

File: packages/ui/src/partials/button.tsx [hunk 1]
```diff
@@ -3,16 +3,20 @@ import { Slot } from "@radix-ui/react-slot";
 import { cva, type VariantProps } from "class-variance-authority";
 
 import { cn } from "../../src/lib/tw-class-merge";
+import { LucideIcon } from "lucide-react";
 
 const buttonVariants = cva(
-  "inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-white transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50",
+  "inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium ring-offset-white transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50",
```
Comment (yaju09) [inline packages/ui/src/partials/button.tsx]:
```suggestion
  "inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium ring-offset-white transition-colors focus-visible:outline-none focus-visible:ring-2  focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50",
```

File: packages/ui/src/partials/button.tsx [hunk 2]
```diff
@@ -3,16 +3,20 @@ import { Slot } from "@radix-ui/react-slot";
...
   {
     variants: {
       variant: {
         default: "grow bg-black text-white hover:bg-black/80",
+        provider: "border text-black hover:bg-black/5",
+        link: "text-black hover:text-black/80",
```
Comment (yaju09) [inline packages/ui/src/partials/button.tsx]:
```suggestion
        link: "text-black hover:text-black/80",
```
will the variant type link be needed for button?

File: packages/ui/src/partials/checkbox.tsx [hunk 3]
```diff
@@ -13,7 +13,7 @@ const Checkbox = React.forwardRef<
   <CheckboxPrimitive.Root
     ref={ref}
     className={cn(
-      "peer h-4 w-4 shrink-0 rounded-sm border border-gray-500 ring-offset-white focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 data-[state=checked]:bg-black/70 data-[state=checked]:text-white",
+      "peer h-4 w-4 shrink-0 rounded-sm border ring-offset-white focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 data-[state=checked]:bg-black/70 data-[state=checked]:rounded-sm data-[state=checked]:text-white",
```
Comment (yaju09) [inline packages/ui/src/partials/checkbox.tsx]:
```suggestion
      "peer h-4 w-4 shrink-0 rounded-sm border ring-offset-white focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 data-[state=checked]:bg-black/70 data-[state=checked]:rounded-sm data-[state=checked]:text-white",
```

File: packages/ui/src/partials/textbox.tsx [hunk 4]
```diff
@@ -9,7 +9,9 @@ const textVariants = cva(
     variants: {
       variant: {
         default: "text-base text-black",
-        description: "text-sm text-gray-400",
+        description: "text-sm text-gray-600",
+        link: "inline text-stone-500",
```
Comment (yaju09) [inline packages/ui/src/partials/textbox.tsx]:
```suggestion
        link: "inline text-stone-500",
```
is variant type link needed for textbox?

**Review by yaju09 — CHANGES_REQUESTED** [review]

**Comment by sidarth-23** [comment]
Closing to resolve conflicts with master

**Pull request closed without merging**